tools/sdk-ai-bots/azure-sdk-qa-bot-backend/Dockerfile

```diff
@@ -1,17 +1,48 @@
+# Stage 1: Build the .NET CLI tool
+FROM mcr.microsoft.com/dotnet/sdk:8.0 AS dotnet-builder
+
+WORKDIR /build
+
+# Copy the CLI project and its dependencies
+COPY tools/azsdk-cli ./tools/azsdk-cli
+COPY tools/codeowners-utils ./tools/codeowners-utils
+
+# Restore and build the CLI tool
+WORKDIR /build/tools/azsdk-cli
+RUN dotnet restore Azure.Sdk.Tools.Cli/Azure.Sdk.Tools.Cli.csproj
+RUN dotnet publish Azure.Sdk.Tools.Cli/Azure.Sdk.Tools.Cli.csproj -c Release -o /azsdk-cli/publish --self-contained false
+
+# Stage 2: Build the Go backend
 FROM mcr.microsoft.com/devcontainers/go:1.24
 
 WORKDIR /app
 
-COPY . .
+# Copy Go application files
+COPY tools/sdk-ai-bots/azure-sdk-qa-bot-backend .
 
+# Build Go application
 RUN go mod download
 RUN go vet -v
 RUN go test -v
-
 RUN CGO_ENABLED=0 go build -o go-backend
 
+# Stage 3: Final runtime image
+FROM mcr.microsoft.com/dotnet/aspnet:8.0
```

Comment:
The dotnet image shouldn't be necessary since the executable is self contained, so the image size could be trimmed down quite a bit.